components/hab/src/main.rs

```diff
@@ -668,7 +668,7 @@ fn sub_pkg_install(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
         LocalPackageUsage::default()
     };
 
-    let install_hook_mode = if m.is_present("IGNORE_INSTALL_HOOK") {
+    let install_hook_mode = if !feat::is_enabled(feat::InstallHook) || m.is_present("IGNORE_INSTALL_HOOK") {
```

Comment:
Since `default` seems more like the happy path, I find it somewhat simpler to think about it in these terms:
```rust
    let install_hook_mode =
        if feat::is_enabled(feat::InstallHook) && !m.is_present("IGNORE_INSTALL_HOOK") {
            InstallHookMode::default()
        } else {
            InstallHookMode::Ignore
        };
```
But the two are equivalent, so either way is fine.